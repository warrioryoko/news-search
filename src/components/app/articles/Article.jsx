import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => (
  <>
    <h1>{title}</h1>
    <p>Written by: {author}</p>
    <p>Article summary: {description}</p>
  </>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Article;
