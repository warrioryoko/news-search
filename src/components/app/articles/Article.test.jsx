import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('does the Article component render?', () => {
  afterEach(() => cleanup());
  it('should render the Article component', () => {
    const { asFragment } = render(<Article
      title="title"
      author="author"
      description="description"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
