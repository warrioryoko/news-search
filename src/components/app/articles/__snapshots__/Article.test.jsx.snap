// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does the Article component render? should render the Article component 1`] = `
<DocumentFragment>
  <h1>
    title
  </h1>
  <p>
    Written by: author
  </p>
  <p>
    Article summary: description
  </p>
</DocumentFragment>
`;
